### PR TITLE
Feature/Add correlation id to json output

### PR DESCRIPTION
**Pull request by sebbi08**

### CODE REVIEW

File: packages/node/src/jsonLayout.ts [hunk 1]
```diff
@@ -1,14 +1,18 @@
 /* eslint-disable @typescript-eslint/naming-convention */
 
+import correlationId from "correlation-id";
 import stringify from "json-stringify-safe";
 import log4js, { LoggingEvent } from "log4js";
 import util from "util";
 
 function jsonFormat(logEvent: LoggingEvent) {
+    // The correlation id is undefined when no set in the current call stack
```
Comment (tnotheis) [inline packages/node/src/jsonLayout.ts]:
```suggestion
    // The correlation id is undefined when not set in the current call stack
```

Comment (tnotheis) [inline packages/node/src/jsonLayout.ts]: Doesn't it generate a new one if it wasn't set explicitly?

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
     "editor.formatOnPaste": true,
     "files.autoSave": "off",
     "editor.codeActionsOnSave": {
-        "source.organizeImports": true
+        "source.organizeImports": "explicit"
```
Comment (jkoenig134) [inline .vscode/settings.json]:
```suggestion
        "source.organizeImports": "always"
```

File: packages/node/src/jsonLayout.ts (hunk 1 above)
Comment (jkoenig134) [reply packages/node/src/jsonLayout.ts]: I think it does and this is not good. We would log an ID that is very misleading.

Comment (sebbi08) [reply packages/node/src/jsonLayout.ts]: Nop at this place it only reads the ID no generation here.

File: packages/node/src/jsonLayout.ts [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
 function jsonFormat(logEvent: LoggingEvent) {
+    // The correlation id is undefined when not set in the current call stack
+    const correlation = correlationId.getId();
     const json = {
         "@t": logEvent.startTime,
         "@mt": util.format(...logEvent.data),
-        "@l": logEvent.level.levelStr
+        "@l": logEvent.level.levelStr,
+        correlationId: correlation
     };
```
Comment (jkoenig134) [inline packages/node/src/jsonLayout.ts]:
```ts
    const json: any = {
        "@t": logEvent.startTime,
        "@mt": util.format(...logEvent.data),
        "@l": logEvent.level.levelStr
    };

    const correlationId = correlationIdLib.getId();
    if (correlationId) json["correlationId"] = correlationId;
```

File: packages/node/src/jsonLayout.ts [hunk 4]
```diff
@@ -1,20 +1,21 @@
...
-    const json = {
+    const json: any = {
         "@t": logEvent.startTime,
         "@mt": util.format(...logEvent.data),
-        "@l": logEvent.level.levelStr,
-        correlationId: correlation
+        "@l": logEvent.level.levelStr
     };
 
+    // The correlation id is undefined when not set in the current call stack
```
Comment (jkoenig134) [inline packages/node/src/jsonLayout.ts]: I removed this comment on purpose, I don't think it is important for the reader to know why it is undefined.

**Review by jkoenig134 — DISMISSED** [review]

**Review by jkoenig134 — APPROVED** [review]

**Review by tnotheis — APPROVED** [review]

**Pull request merged**